### DISCUSSION TITLE
"ctrl + option + ↑ + →" for Top Right

### DISCUSSION THREAD

**Discussion by 1ennard** [rxhanson/Rectangle#1335] [Ideas]
It would be great if 2 arrow keys could be combined to move the window in the corners.
Right now, a max of 3 keys can be combined, even with the option _Remove keyboard restrictions_ enabled.

I would like to create these Shortcuts:
"ctrl + option + ↑ + →" for Top Right
"ctrl + option + ↑ + ←" for Top Left
"ctrl + option + ↓ + →" for Bottom Right
"ctrl + option + ↓ + ←" for Bottom Left

**Comment by rxhanson** [comment]
The keyboard shortcut API in macOS defines keyboard shortcuts as modifier keys + 1 other key. In order to get around this limitation, Rectangle would have to listen to keystrokes. I don't have plans to add this in myself, but see no reason to turn down a good pull request that adds this.

**Comment by dg-pb** [comment]
++ For this. I am transitioning from `hyperdock`, and missed this. It is shortcut space saving feature.

**Reply by mbcullen** [reply]
I'm another former HyperDock user that really misses this functionality. I was beginning to wonder if I had imagined the whole thing since it seems so obvious, but no other window manager offers the capability. I'd love to see native support but found an alternative. You can use a keyboard customizer like Karabiner Elements to remap your shortcuts. It's a little annoying to have the additional layer in the middle and potential for failure, but it's worth it in my experience so far.

**Comment by rrshaban** [comment]
JFYI, this seems covered today by ctrl_option_u, i, j, k

**Reply by rxhanson** [reply]
Even though the corner quadrants are covered by those shortcuts, it's not quite the same as having multi-key shortcuts, which are requested from time to time. I rebuilt the shortcut recorder that Rectangle uses (MASShortcut) for Rectangle Pro to accommodate a lot of other shortcut functionality, where I am planning on adding in multi-key shortcuts, and when it's ready I'm planning on open sourcing it and using it in Rectangle.

**Comment by mbcullen** [comment]
That’s right. I know it’s possible to cover the quadrants with other shortcuts, but the arrow keys are much easier to remember.

I ended up using Karabiner to translate between the shortcuts I wanted to use and Rectangle-compatible ones.

**Comment by dg-pb** [comment]
Also, it is good to keep to arrows - avoids conflicts with other shortcuts (like I found for ctrl_option_u, i, j, k). This is the one and only feature that I miss from my old window manager.
